extension/src/entrypoints/append-test-case.ts

```diff
@@ -0,0 +1,102 @@
+import { ExtractMessage, MessagePayload, WindowMessage } from "@cb/types";
+import {
+  waitForElementAtIndex,
+  waitForElementsWithCondition,
+} from "@cb/utils/dom";
+import { assertUnreachable } from "@cb/utils/error";
+import { defineUnlistedScript } from "wxt/utils/define-unlisted-script";
+
+const appendTestCaseToLeetCode = async (
+  args: MessagePayload<
+    ExtractMessage<WindowMessage, "appendTestCaseToLeetCode">
+  >
+) => {
+  const { testValues } = args;
+
+  try {
+    const testCaseButtons = document.querySelectorAll(
+      'button[data-e2e-locator="console-testcase-tag"]'
+    );
+
+    if (testCaseButtons.length === 0) {
+      throw new Error("No test case buttons found");
+    }
+
+    const lastTestCaseButton = testCaseButtons[
+      testCaseButtons.length - 1
+    ] as HTMLElement;
+
+    const addButton = lastTestCaseButton.parentElement?.querySelector(
```

Comment:
Let's just collapse the entire thing

```suggestion
const addButton = document.querySelectorAll(
      'button[data-e2e-locator="console-testcase-tag"]'
    )[testCaseButtons.length - 1]?.parentElement?.querySelect(...);
```